lib/TokenManager.js

```diff
@@ -122,17 +143,39 @@ function refresh(sdk, tokenMgmtRef, storage, key) {
   // Remove existing autoRefresh timeout for this key
   clearRefreshTimeout(tokenMgmtRef, key);
 
+  var deferred = Q.defer();
+
+  if (!tokenMgmtRef.refreshPromises[key]) {
+    tokenMgmtRef.refreshPromises[key] = [];
+  }
+
+  tokenMgmtRef.refreshPromises[key].push(deferred);
+
+  if (!tokenMgmtRef.refreshPromise[key]) {
+    tokenMgmtRef.refreshPromise[key] = refreshToken(sdk, tokenMgmtRef, storage, key, token);
+  } else {
+    deferred.resolve();
```

Comment:
We don't want to resolve this now, we want to let `resolveRefreshTimeouts()` do this later.